ghost/core/core/server/data/importer/importers/data/custom-theme-settings.js

```diff
@@ -18,64 +17,57 @@ class CustomThemeSettingsImporter extends BaseImporter {
-                                .then((importedModel) => {
-                                    if (importOptions.returnImportedData) {
-                                        this.importedDataToReturn.push(importedModel.toJSON());
-                                    }
-                                    return importedModel;
-                                })
-                                .catch((err) => {
-                                    return this.handleError(err, item);
-                                });
-                        }
+                if (setting) {
+                    setting.set('value', item.value);
+                }
 
-                        return Promise.resolve();
-                    }
+                if (setting && !setting.hasChanged()) {
+                    // No change to make, pop next entry and continue
+                    item = this.dataToImport.shift();
+                    continue;
+                }
 
-                    return models.CustomThemeSetting.add(item, options)
-                        .then((importedModel) => {
-                            if (importOptions.returnImportedData) {
-                                this.importedDataToReturn.push(importedModel.toJSON());
-                            }
-                            return importedModel;
-                        })
-                        .catch((err) => {
-                            return this.handleError(err, item);
-                        });
-                })
-                .reflect());
-        });
-        
-        const opsPromise = Promise.all(ops);
+                const importedModel = setting
+                    ? await setting.save(null, options)
+                    : await models.CustomThemeSetting.add(item, options);
+
+                if (importOptions.returnImportedData) {
+                    this.importedDataToReturn.push(importedModel.toJSON());
+                }
+            } catch (error) {
+                if (error) {
+                    importErrors.push(...this.handleError(error, item));
+                }
+            }
+
+            // Shift next entry
+            item = this.dataToImport.shift();
+        }
 
-        // activate function is called to refresh cache when importing custom theme settings for active theme
-        opsPromise.then(() => {
-            models.Settings.findOne({key: 'active_theme'})
-                .then((theme) => {
-                    const currentTheme = theme.get('value');
-                    if (this.dataToImport.some(themeSetting => themeSetting.theme === currentTheme)) {
-                        activate(currentTheme);
-                    }
-                });
+        models.Settings.findOne({key: 'active_theme'}).then((theme) => {
+            const currentTheme = theme.get('value');
+            if (this.dataToImport.some(themeSetting => themeSetting.theme === currentTheme)) {
```

Comment:
@sam-lord since you're using `this.dataToImport.shift()` above, this will always become falsy. Haven't checked other files. Saw this because I was working on this file on another PR.

Reply:
Thanks for noticing this! I've fixed this by checking the condition first, then activating the theme if the condition is true after importing all of the data.

I've checked all other importers and custom-theme-settings was the only one which referenced the dataToImport array during import - so this should be fixed now.